### ISSUE TITLE
Passenger Apps not working

### ISSUE THREAD

**Issue by btsherid** [OSC/ondemand#3082]
Hi,

I have two OnDemand servers, both running 3.0.1. One is CentOS 7 and one is Rocky 9. 

On the CentOS 7 install, the passenger app works fine and displays a table. I copied the app to the Rocky 9 host and it does not error, but it does not display a table. It looks like maybe the loop in views/index.erb is not actually pulling data from the struct?

Working app on Centos 7:
![image](https://github.com/OSC/ondemand/assets/62118099/eefc73f1-09b3-4407-bd58-6169f5b687cf)

What I get on Rocky 9 with the same app:

![image](https://github.com/OSC/ondemand/assets/62118099/740d0492-3a00-4d55-8d1d-846fe562c2aa)

**Comment by treydock** [comment]
Does that sacct command on Rocky 9 return expected and same output as CentOS 7 when run as the user you logged in as?

**Comment by btsherid** [comment]
Yes, I ran a diff between the CentOS 7 and Rocky 9 outputs and they are the same.

**Comment by btsherid** [comment]
So it appears that the Passenger app is running the code as the apache user. As the apache user, SLURM commands don't work which is why the table output is empty. 

Is this expected behavior and I need to investigate why SLURM is not working for the apache user, or should this code be running as the user that is logged into OnDemand (me)?

Thanks,
Brendan

**Comment by johrstrom** [comment]
> I have fixed this by updating command.rb in each application, but I'm wondering if it is expected behavior for the command to run as the apache user instead of the user logged into OnDemand.

It should not be running as the apache user. How did you deploy this application?

**Comment by btsherid** [comment]
I made a new app through the Developer menu and providing the Git url as explained in the documentation. I also tried just copying the folder from another host running the same version of OnDemand. Both methods worked.

I did some more debugging and it is running as my user. It does not have all the environment variables it needs. That's what I had to add to the command. This makes sense to me because I installed the version of SLURM from source. I'm planning on upgrading SLURM soon so it shouldn't be an issue going forward. I'll just need the enviornment variable in the commands for now.

Thanks,
Brendan
